feat: Disable language analysis for a repository DOCS-686 (#2112)
codacy/docs: docs/repositories-configure/codacy-configuration-file.md

    ignoreAnnotations: false
    ignoreUsings: false
```

## Configuring languages using a Codacy configuration file {: id="languages"}

You can use a Codacy configuration file to manage the languages that Codacy analyzes in your repository.

!!! note

>     Codacy applies both language settings defined in the Codacy configuration file and any other language setting defined [on the Codacy UI](languages.md).

Author: > Note: it's also unclear "how" these settings are applied. Which ones take precedence?

@nicklem AFAIK, there's no precedence, they are cumulative:
- If we have different custom extensions defined in the configuration file and on the Codacy UI, all of them will be applied. 
- All languages [are enabled by default](https://github.com/codacy/docs/pull/2112/files#diff-6e1b2afc217b0798259f3b0065e205a68fd5c13f20e03dae4014a2c39509173aR279). Thus, Codacy will not analyze the languages disabled in the configuration file, nor the languages disabled on the Codacy UI. Makes sense? 🤔 

@pedrocodacy Can you confirm if there is no precedence for these configurations?

>     Codacy applies both language settings defined in the Codacy configuration file and any other language setting defined [on the Codacy UI](languages.md).

Author: @nicklem Did you remove the link for a particular reason? Let me know if this works:

```suggestion
    Codacy applies the language settings from the Codacy configuration file as well as any settings defined [in the Codacy UI](languages.md).
```

>     Codacy applies both language settings defined in the Codacy configuration file and any other language setting defined [on the Codacy UI](languages.md).

Reviewer: Thanks @claudiacarpinteiro, all clear!

> Did you remove the link for a particular reason?
I believe the autocorrect removed it 😅 let's keep it, yes

>     Codacy applies both language settings defined in the Codacy configuration file and any other language setting defined [on the Codacy UI](languages.md).

Author: Keeping your suggestion and avoid "both".

```suggestion
    Codacy applies the language settings from the Codacy configuration file as well as any settings defined [in the Codacy UI](languages.md).
```
